ibm/service/vmware/data_source_ibm_vmaas_transit_gateway_connection.go

```diff
@@ -0,0 +1,278 @@
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Determines the state of the connection.",
+						},
+						"local_gateway_ip": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Local gateway IP address for the connection.",
+						},
+						"remote_gateway_ip": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Remote gateway IP address for the connection.",
+						},
+						"local_tunnel_ip": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Local tunnel IP address for the connection.",
+						},
+						"remote_tunnel_ip": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Remote tunnel IP address for the connection.",
+						},
+						"local_bgp_asn": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Local network BGP ASN for the connection.",
+						},
+						"remote_bgp_asn": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Remote network BGP ASN for the connection.",
+						},
+						"network_account_id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the account that owns the connected network.",
+						},
+						"network_type": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The type of the network that is connected through this connection. Only \"unbound_gre_tunnel\" is supported.",
+						},
+						"base_network_type": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The type of the network that the unbound GRE tunnel is targeting. Only \"classic\" is supported.",
+						},
+						"zone": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The location of the connection.",
+						},
+					},
+				},
+			},
+			"status": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Determines the state of the IBM Transit Gateway based on its connections.",
+			},
+			"region": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The region where the IBM Transit Gateway is deployed.",
+			},
+		},
+	}
+}
+
+func dataSourceIbmVmaasTransitGatewayConnectionRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vmwareClient, err := meta.(conns.ClientSession).VmwareV1()
+	if err != nil {
+		tfErr := flex.DiscriminatedTerraformErrorf(err, err.Error(), "(Data) ibm_vmaas_transit_gateway_connection", "read", "initialize-client")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	vdcIDRaw, ok := d.GetOk("vdc_id")
```

Comment:
If we declared as Required in Schema we explictly don't need these (	if !ok || vdcIDRaw == nil {)
Terraform will handle it

Reply:
ok